conn.go

```diff
@@ -156,6 +156,36 @@ func ConnContainerID(id string) ConnOption {
 	}
 }
 
+// ConnServer indicates that the underlying net.Conn is a server connection.
+//
+// This option only relevant if you use amqp.New() to create a Client from an existing net.Conn.
+// amqp.Dial() always creates a client connection.
+// amqp.Accept() always creates a server connection.
+func ConnServer() ConnOption { return nil }
+
+// IMPLEMENTATION NOTE: ConnServer is needed because the SASL negotiation is asymmetric.
+// Everything after that (including the AMQP OPEN frame) is symmetric so almost
+// no change is required to frame-processing code except to support incoming endpoints.
+
+// Implementation note - the ConnServer option is needed for the asymmetric
+// SASL negotiation. Once that is done, AMQP is perfectly symmetric.
+
+// ConnSession sets a callback for incoming session begin.
+// To reject the session, call Detach().
```

Comment:
Why Detach() and not e.g. End() to match protocol (or Close() as a more common generic shutdown action).

Reply:
Agreed. 

Renamed to CloseErr(). The API already offers a uniform Close() for orderly close so CloseErr() fits the existing style.

Made some other changes too: I replaced the multiple handler options with a single handler interface. This fixes a pitfall in electron's API where you want to accept Receivers but forget that you also have to accept Sessions or you'll never get a Receiver. Now you have to implement a Handler that explicitly says what to do for all 3 cases.

The new Handler gets an IncomingSender (etc.) object containing the begin/attach performative data *not* a usable *Sender object. It can call IncomingSender.Accept() to accept and get a usable *Sender, or return amqp.Error to reject. 

I thought I could blur that distinction for simplicity but on reflection I can't, otherwise it's too easy for applications to attempt to Send() on a Sender that isn't open yet, with hilarious consequences (not). This would create a nasty race condition  when handlers spin up goroutines to  handle endpoints concurrently, unless we have a clear point at which we say "ok now you can use it" - which is what the Accept() call is for.